video samples fail to build

The following samples fail to build:
```
samples/subsys/video/capture/sample.video.capture in mimxrt1064_evk:samples/subsys/video/capture/sample.video
samples/subsys/video/tcpserversink/sample.video.tcpserversink in mimxrt1064_evk:samples/subsys/video/tcpserversink/sample.video
```

Failure is of form:

```
/workdir/zephyr/drivers/video/video_sw_generator.c: In function 'video_sw_generator_stream_stop':
/workdir/zephyr/drivers/video/video_sw_generator.c:66:2: error: too few arguments to function 'k_work_cancel_delayable_sync'
   66 |  k_work_cancel_delayable_sync(&data->buf_work);
      |  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../../../../../../../include/zephyr.h:18,
                 from /workdir/zephyr/drivers/video/video_sw_generator.c:6:
../../../../../../../include/kernel.h:3363:6: note: declared here
 3363 | bool k_work_cancel_delayable_sync(struct k_work_delayable *dwork,
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/drivers/video/video_sw_generator.c
+++ b/drivers/video/video_sw_generator.c
@@ -16,6 +16,7 @@ struct video_sw_generator_data {
 	struct k_fifo fifo_in;
 	struct k_fifo fifo_out;
 	struct k_work_delayable buf_work;
+	struct k_work_sync work_sync;
 	int pattern;
 	bool ctrl_hflip;
 	bool ctrl_vflip;
@@ -63,7 +64,7 @@ static int video_sw_generator_stream_stop(const struct device *dev)
 {
 	struct video_sw_generator_data *data = dev->data;
 
-	k_work_cancel_delayable_sync(&data->buf_work);
+	k_work_cancel_delayable_sync(&data->buf_work, &data->work_sync);
 
 	return 0;
 }

